Sources/Sentry/SentryProfiler.mm

```diff
@@ -99,20 +99,60 @@ - (void)start
         }
         _profile = [NSMutableDictionary<NSString *, id> dictionary];
         const auto sampledProfile = [NSMutableDictionary<NSString *, id> dictionary];
+
+        /*
+         * Maintain an index of unique frames to avoid duplicating large amounts of data. Every
+         * unique frame is stored in an array, and every time a stack trace is captured for a
+         * sample, the stack is stored as an array of integers indexing into the array of frames.
+         * Stacks are thusly also stored as unique elements in their own index, an array of arrays
+         * of frame indices, and each sample references a stack by index, to deduplicate common
+         * stacks between samples, such as when the same deep function call runs across multiple
+         * samples.
```

Comment:
💯 thanks for the detailed explanation. Would something like this also make sense for SentryCrash? I guess not because we don't have duplicate frames that often.

Reply:
yeah not the same use case